cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/DynamicDatasetContext.java

```diff
@@ -38,24 +43,45 @@
   private final Set<String> allowedDatasets;
   private final DatasetFramework datasetFramework;
   private final ClassLoader classLoader;
+  private final Map<String, String> runtimeArguments;
 
   @Nullable
   protected abstract LoadingCache<Long, Map<String, Dataset>> getDatasetsCache();
 
+  /**
+   * Get the runtime arguments for a specific dataset. Arguments not in the scope of the dataset are filtered out.
+   * For example, when getting the runtime arguments for a dataset named 'myds', only runtime arguments that start
+   * with 'dataset.myds.' are returned, with the prefix stripped from the arguments.
+   *
+   * @param name the name of the dataset
+   * @return runtime arguments for the given dataset
+   */
+  protected Map<String, String> getRuntimeArguments(String name) {
+    return RuntimeArguments.extractScope(Scope.DATASET, name, runtimeArguments);
+  }
+
   public DynamicDatasetContext(TransactionContext context, DatasetFramework datasetFramework, ClassLoader classLoader) {
-    this(context, datasetFramework, classLoader, null);
+    this(context, datasetFramework, classLoader, null, ImmutableMap.<String, String>of());
```

Comment:
minor: since immutable, this could be a static final constant.  doesn't matter that much though.